plugins/inputs/system/ps.go

```diff
@@ -73,35 +73,27 @@ func (s *systemPS) DiskUsage(
 	var usage []*disk.UsageStat
 	var partitions []*disk.PartitionStat
 
-	for i := range parts {
-
-		p := parts[i]
+	for _, p := range parts {
+		// If the mount point is not a member of the filter set,
+		// don't gather info on it.
+		if _, ok := mountPointFilterSet[p.Mountpoint]; !ok {
+			continue
+		}
 
-		if len(mountPointFilter) > 0 {
-			// If the mount point is not a member of the filter set,
-			// don't gather info on it.
-			_, ok := mountPointFilterSet[p.Mountpoint]
-			if !ok {
-				continue
-			}
+		// If the mount point is a member of the exclude set,
+		// don't gather info on it.
+		if _, ok := fstypeExcludeSet[p.Fstype]; ok {
+			continue
 		}
+
 		mountpoint := os.Getenv("HOST_MOUNT_PREFIX") + p.Mountpoint
 		if _, err := os.Stat(mountpoint); err == nil {
-			du, err := disk.Usage(mountpoint)
-			if err != nil {
-				return nil, nil, err
-			}
-			du.Path = p.Mountpoint
-
-			// If the mount point is a member of the exclude set,
-			// don't gather info on it.
-			_, ok := fstypeExcludeSet[p.Fstype]
-			if ok {
-				continue
+			if du, err := disk.Usage(mountpoint); err == nil {
+				du.Path = p.Mountpoint
+				du.Fstype = p.Fstype
```

Comment:
I meant I see Path has the same value as Mountpoint.
Also Fstype field in "p" has the same value as in "du".
p is of disk.PartitionStat type and du is disk.UsageStat : https://godoc.org/github.com/shirou/gopsutil/disk#PartitionStat

That's why I don't see why are they assigned.
Unless I'm missing something I can update the PR to remove those two assignments if they are indeed redundant.